### ISSUE TITLE
How to use authentication

### ISSUE THREAD

**Issue by DaveDavenport** [yukibtc/ntfy-sdk#1]
I am trying to use this lib with my own ntfy server that requires authentication.

I tried changing the example by passing 'https://user:pass@host' to Dispatcher, but that did not work.

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ReqwestError(reqwest::Error { kind: Builder, source: reqwest::Error { kind: Builder, source: RelativeUrlWithoutBase } })', src/main.rs:5:79
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::result::unwrap_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/result.rs:1785:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/result.rs:1107:23
   4: mqtt2ntfy::main::{{closure}}
             at ./src/main.rs:5:22
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/future/mod.rs:91:19
   6: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at /home/-/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/park.rs:272:63
   7: tokio::runtime::coop::with_budget
             at /home/-/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/coop.rs:102:5
   8: tokio::runtime::coop::budget
             at /home/-/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/coop.rs:68:5
   9: tokio::runtime::park::CachedParkThread::block_on
             at /home/-/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/park.rs:272:31
  10: tokio::runtime::context::BlockingRegionGuard::block_on
             at /home/-/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/context.rs:255:13
  11: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at /home/-/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/scheduler/multi_thread/mod.rs:66:9
  12: tokio::runtime::runtime::Runtime::block_on
             at /home/-/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.22.0/src/runtime/runtime.rs:281:45
  13: mqtt2ntfy::main
             at ./src/main.rs:11:5
  14: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
```

**Comment by yukibtc** [comment]
Hi, I just released version 0.2.0 that include support to authentication.

**Comment by DaveDavenport** [comment]
Thanks, that worked!
